src/nccl_ofi_net.c

```diff
@@ -2477,6 +2594,16 @@ static ncclResult_t ofi_deregMr(void *comm, void *mhandle)
 				rc, fi_strerror(-rc));
 	}
 
+	if (!prov_key_mr) {
+		uint64_t key = fi_mr_key(mr_handle);
```

Comment:
Do we need to close `mr_handle` with the provider if these are created by the plugin?

Reply:
There's nothing extra to close, OFI should clean this up when we call `fi_close` on `mr_handle`. However, I noticed that we call `fi_mr_key` on `mr_handle` after closing the handle, which is an error. I'll fix the order of these calls.